src/Gateways/GpApiGateway.php

```diff
@@ -550,6 +567,40 @@ public function handle_adding_capture_order_action( $actions ) {
 		return $actions;
 	}
 
+	/**
+	 * Handles subscription renewals, gets the muti-use token from the order meta and charges it. 
+	 * @return bool
+	 */
+
+	public function renew_subscription($amount_to_charge, $renewal_order)
+	{
+		$renewal_order_subscriptions = wcs_get_subscriptions_for_renewal_order($renewal_order->get_id(),array("order_type"=>"parent"));
+		$parent_order = false;
+		foreach($renewal_order_subscriptions as $renewal_order_subscription){
+			if($renewal_order_subscription->get_parent_id()){
+				$parent_order = wc_get_order($renewal_order_subscription->get_parent_id());
+			}
+		}
+
+		if( !$parent_order || !$parent_order->get_meta("_GP_multi_use_token")){
+			return;
+		}
+		$request  = $this->prepare_request(parent::TXN_TYPE_SUBSCRIPTION_PAYMENT, $renewal_order, array("multi_use_token"=> $parent_order->get_meta("_GP_multi_use_token")));
+		$response = $this->client->submit_request( $request );
+		$client_trans_ref = $response->transactionReference->clientTransactionId;
+		if(parent::handle_response($request,$response)){
+			$renewal_order->add_order_note(__("Subscription Renewal Successful \r\n Transaction Reference: ". $client_trans_ref));
```

Comment:
For adding variables to translatable strings you will need to use the 'sprintf' function. You are also missing the translation domain.
Please check other strings translated in the plugin.
https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/